lib/controllers/users.js

```diff
@@ -0,0 +1,178 @@
+    siteId: '-1',
+    activeFlag: 'Y',
+    passwordExpDate: moment().add(1, 'years').format('MM/DD/YYYY'),
+    acctExpDate: moment().add(1, 'years').format('MM/DD/YYYY')
+};
+
+/**
+ * Call the toJSON method on the object
+ * @param  {Object} obj object to call the toJSON method on (bookshelf)
+ * @return {string}     JSON
+ */
+internals.callToJson = (obj) => {
+    return obj.toJSON();
+};
+
+/**
+ * Call the fetch method on the object
+ * @param  {Object} obj object to call the fetch method on (bookshelf)
+ * @return {Promise}
+ */
+internals.callFetch = (obj) => {
+    //remove all attributes but username
+    obj.attributes = { username: obj.get('username') };
+    return obj.fetch();
+};
+
+/**
+ * hide protected fields so that they are no returned
+ * @param  {object} obj bookshelf object of user
+ * @return {object}     sanitized model
+ */
+internals.sanitize = (obj) => {
+    delete obj.attributes.passwordHash;
+    delete obj.attributes.passwordResetKey;
+    return obj;
+};
+
+/**
+ * Add new user to database
+ * @param  {object} userModel bookshelf model to fetch user data
+ * @param  {object} userData  object containing properties to match the user on
+ * @return {[type]}           [description]
+ */
+internals.addUser = (authUtils, UserModel, userData) => {
+    const userDataClone = _.clone(userData);
+    const username = userDataClone.username;
+    const password = userDataClone.password;
+    const setHashedPassword = (password) => {
+        userDataClone.passwordHash = password;
+        userDataClone.password = null;
+        return userDataClone;
+    };
+
+    const saveUser = (userData) => {
+        return UserModel.forge(userData).save(null, {method: 'insert'});
+    };
+
+    if (username) {
+        userDataClone.username = authUtils.encryptUsername(username);
+    }
+
+    return authUtils.hashPassword(password)
+        .then(setHashedPassword)
+        .then(_.partialRight(_.defaults, internals.defaultValues))
+        .then(saveUser);
+};
+
+module.exports.create = {
+    tags: ['api', 'users'],
+    notes: [
+        'Creates a new user with the properties in the payload.'
+    ].join('<br>'),
+    description: 'Creates a new user with the properties in the payload.',
+    auth: false,
+    validate: {
+        payload: joi.object().keys({
+            username: joi.string().min(3).max(20).required(),
+            email: joi.string().email().required(),
+            label: joi.string().required(),
+            password: joi.string().required().min(8).max(32)
+        })
+    },
+    response: {
+        schema: joi.object().keys({
+            username: joi.string().min(3).max(20).required(),
+            email: joi.string().email().required(),
+            label: joi.string().required(),
+            siteId: joi.string().required(),
+            activeFlag: joi.string().required(),
+            passwordExpDate: joi.any().required(),
+            acctExpDate: joi.any().required(),
```

Comment:
It doesn't work. We get something strange back from the DB... I was in too much of a hurry to investigate why, but will look into it further now.